Fix expiry date related bugs

songyiang opened the pull request:
Error messages for invalid expiry dates have been standardised.

The year of expiry date is now constrained between 2020 and 2120, both
inclusive.

More test cases for expiry date have been added.

Commit:
Fix expiry date related bugs

Error messages for invalid expiry dates have been standardised.

The year of expiry date is now constrained between 2020 and 2120, both
inclusive.

More test cases for expiry date have been added.

bchenghi commented on src/main/java/seedu/simplykitchen/model/food/ExpiryDate.java:
```diff
@@ -16,9 +16,8 @@
-    public static final String MESSAGE_SHORTENED_YEAR = "The year should be 4 digits long. ";
-    public static final String MESSAGE_INVALID_DATE = "The expiry date does not exist. ";
+            "The expiry date should be of the format DD-MM-YYYY or DD/MM/YYYY. "
+                    + "The year should be between 2020 and 2120, both inclusive.";
```
Should we include an error message for invalid date? For example 29-2-2021 is the correct format and is between 2020 and 2120, but is not valid.

songyiang replied:
I edited the message to include the valid date part.

The two date formatters used check for different types of formatting errors and also invalid date as well. Hence, the formatters can't differentiate between an invalid date or a date with incorrect formatting.

bchenghi replied:
Oh I see. Thanks!

bchenghi reviewed (commented):
The changes look good! I added one comment.

Commit:
Merge branch 'master' into v1.4-expiry-date

Commit:
Fix error message for invalid expiry date

The error message asks user to input a valid date if an invalid date
with the correct formatting is enterd e.g. 29-02-2021.

bchenghi reviewed (approved):
LGTM
